index.js

```diff
@@ -190,7 +191,19 @@ function globIsSingular(glob) {
 }
 
 function getBasePath(ourGlob, opt) {
-  return resolveGlob(globParent(ourGlob) + path.sep, opt);
+  var basePath;
+  var parent = globParent(ourGlob);
+
+  if (parent === '/' && opt && opt.root) {
+    basePath = path.join(opt.root);
+  } else {
+    basePath = resolveGlob(parent, opt);
+  }
+
+  if (!sepRe.test(basePath.charAt(basePath.length - 1))) {
```

Comment:
> search the whole string

I thought that too, but [this talk](https://vimeo.com/112065252) is where I learned more about regexp. I'm assuming that it works this way when using `$` too because the regexp compiler would need match patterns leading up to the `$`.

If I find the time, I may put together some benchmarks with these patterns.